### PR TITLE
Move Page Iterator to core

### LINKED ISSUE #179: PageIterator should be shared in core - moved out of service library
[AB#6802](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6802)

### PR DESCRIPTION
**Pull request by andrueastman**
Closes #179 

It involves moving the code in the PR below and modifying the tests to run in core.

Related to https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/977 (to be merged in together)
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/246)

### CODE REVIEW

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionRequest.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Graph.DotnetCore.Core.Test.Requests;
+
+namespace Microsoft.Graph.DotnetCore.Test.Tasks
+{
+    public class MockUserEventsCollectionRequest : TestEventDeltaRequest
+    {
+        private TestEventDeltaCollectionPage nextPage;
```
Comment (baywet) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionRequest.cs:16]: could we replace this class by a mocking framework?

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionPage.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.Graph.DotnetCore.Core.Test.TestModels.ServiceModels;
+
+namespace Microsoft.Graph.DotnetCore.Test.Mocks
+{
+    using Microsoft.Graph.DotnetCore.Test.Tasks;
+    using System;
+    using System.Collections.Generic;
+    public class MockUserEventsCollectionPage : CollectionPage<TestEvent>, ITestEventDeltaCollectionPage
+    {
+        public MockUserEventsCollectionPage(IList<TestEvent> currentPage, TestEventDeltaRequest nextPageRequest, string linkType = "") : base(currentPage)
```
Comment (baywet) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionPage.cs:14]: could we replace this class by a mocking framework?

**Review by baywet — APPROVED** [review]

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionPage.cs (hunk 2 above)
Comment (andrueastman) [reply tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionPage.cs:14]: Will investigate this.

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionRequest.cs (hunk 1 above)
Comment (andrueastman) [reply tests/Microsoft.Graph.DotnetCore.Core.Test/Mocks/MockUserEventsCollectionRequest.cs:16]: Will investigate this.

**Review by baywet — APPROVED** [review]

**Pull request merged**